apps/test/unit/code-studio/pd/form_components_func/FormControllerTest.js

```diff
@@ -20,594 +21,567 @@ let DummyPage3 = () => {
-            previouslySaved
-              ? setUpValidWithApplicationId(applicationId)
-              : setupValid();
-
-            triggerSubmit();
-
-            expect(getErrors(DummyPage3)).to.be.empty;
-            expect(server.requests).to.have.length(1);
-            expect(server.requests[0].method).to.eql(
-              previouslySaved ? 'PUT' : 'POST'
-            );
-            expect(server.requests[0].url).to.eql(
-              previouslySaved
-                ? `${fakeEndpoint}/${applicationId}`
-                : fakeEndpoint
-            );
-          });
+          expect(getErrors(DummyPage3)).to.be.empty;
+          expect(server.requests).to.have.length(1);
+          expect(server.requests[0].method).to.eql(
+            previouslySaved ? 'PUT' : 'POST'
+          );
+          expect(server.requests[0].url).to.eql(
+            previouslySaved ? `${fakeEndpoint}/${applicationId}` : fakeEndpoint
+          );
         });
+      });
 
-        it('Disables the submit button during submit and renders spinner', () => {
-          setupValid();
-          triggerSubmit();
-          expect(form.findOne('#submit').props.disabled).to.be.true;
-          expect(form.findAll('Spinner')).to.have.length(1);
-        });
+      it('Disables the submit button during submit and renders spinner', () => {
+        setupValid();
+        triggerSubmit();
+        expect(form.findOne('#submit').props).to.be.disabled;
+        expect(form.findAll('Spinner')).to.have.length(1);
+      });
 
-        it('Re-enables the submit button on error and removes spinner', () => {
+      [
+        serverResponse(400, {
+          errors: {form_data: ['an error']}
+        }),
+        serverResponse(500)
+      ].forEach(response => {
+        const statusNumber = response[0];
+        it(`Re-enables the submit button on ${statusNumber} error and removes spinner`, () => {
           setupValid();
-          server.respondWith([
-            400,
-            {'Content-Type': 'application/json'},
-            JSON.stringify({
-              errors: {form_data: ['an error']}
-            })
-          ]);
+          server.respondWith(response);
 
           triggerSubmit();
           server.respond();
 
-          expect(getErrors(DummyPage3)).to.eql(['an error']);
-          expect(form.findOne('#submit').props.disabled).to.be.false;
+          statusNumber === 400 &&
+            expect(getErrors(DummyPage3)).to.eql(['an error']);
+          expect(form.findOne('#submit').props).to.be.disabled;
           expect(form.findAll('Spinner')).to.have.length(0);
         });
+      });
 
-        // [MEG] TODO: Refactor this to use savedStatus instead of form_data
-        it('Overrides application status as unreviewed on submit', () => {
-          const testData = {
-            field1: 'value 1',
-            field2: 'value 2',
-            status: 'incomplete'
-          };
+      it('Sends application status as unreviewed on submit', () => {
```

Comment:
Refactored to address `// [MEG] TODO: Refactor this to use savedStatus instead of form_data`